counter_contract_interaction_script/project/tests/src/harness.rs

```diff
@@ -0,0 +1,66 @@
+use fuels::{prelude::*, accounts::wallet::WalletUnlocked};
+
+abigen!(
+    Contract(
+        name = "CounterContract",
+        abi = "./contract/out/debug/counter_contract-abi.json"
+    ),
+    Script(
+        name = "InteractionScript",
+        abi = "./script/out/debug/interaction_script-abi.json" 
+    )
+);
+
+pub async fn setup() -> (CounterContract<WalletUnlocked>, WalletUnlocked) {
+    let number_of_wallets = 1;
+    let coins_per_wallet = 1;
+    let amount_per_coin = 1_000_000_000;
+
+    let config = WalletsConfig::new(
+        Some(number_of_wallets),
+        Some(coins_per_wallet),
+        Some(amount_per_coin),
+    );
+
+    let mut wallets = launch_custom_provider_and_get_wallets(config, None, None).await;
+
+    let wallet = wallets.pop().unwrap();
+
+    let storage_configuration = StorageConfiguration::load_from("../contract/out/debug/counter_contract-storage_slots.json");
+    let configuration =
+        LoadConfiguration::default().set_storage_configuration(storage_configuration.unwrap());
+
+    let id = Contract::load_from("../contract/out/debug/counter_contract.bin", configuration)
+        .unwrap()
+        .deploy(&wallet, TxParameters::default())
+        .await
+        .unwrap();
+
+    let instance = CounterContract::new(id, wallet.clone());
+
+    (instance, wallet)
+}
+
+#[tokio::test]
+async fn test_script_clearing_at_end() {
```

Comment:
These tests aren't particularly useful.

1) I would rename
2) I would split the script into 2, increment and then clear so that you can call one after the other. It may not make sense as you can pass in a variable but ideally when writing code you want things to remain invariant and reusable. That way you can select 1 type of script over another. Variable complexity can be added later but for simplicity it's also better to separate it here, after all, your description mentions calling contracts (instead of scripts) in any order and here we only do it once.

Reply:
I added the clear bool parameter on purpose to demonstrate that scripts can have conditional execution, and that outputs can differ if inputs are changed

Reply:
The counter is too simple to have 3 scripts

- Increment
- Clear
- Use arguments

In terms of writing code that is easy to analyze and verify I still opt for my approach but at the same time having an explicit example of a script which uses args is also useful.

So here is my proposition:

- Leave it as it is and perhaps later I will extend the counter to have scripts with args and without
- You may do this yourself now as per my suggestion

Up to you

Reply:
This is gonna be used in the quickstart so the arg feature needs to demonstrated, i will leave it as is